lib/manager/components/transform/AddCustomFile.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+
+import React, { Component } from 'react'
+
+import type { AddCustomFileProps, TransformProps } from '../../../types'
+import CSV_VALIDATION_ERRORS from '../../util/enums/transform'
+import { getComponentMessages } from '../../../common/util/config'
+
+import CustomCSVForm from './CustomCSVForm'
+
+/**
+ * Component that renders fields for AddCustomFile. This transformation shares csvData props with the ReplaceFileFromString transformation.
+ * TODO: adapt this transformation to include a file upload for larger custom files?
+ */
+export default class AddCustomFile extends Component<TransformProps<AddCustomFileProps>, AddCustomFileProps> {
+  // Messages are for the child CSV Form component but since messages are shared across transformation types,
+  // the messages are being grouped under that component.
+  messages = getComponentMessages('AddCustomFile')
+  constructor (props: TransformProps<AddCustomFileProps>) {
+    super(props)
+    this.state = {csvData: props.transformation.csvData, table: props.transformation.table}
+  }
+
+  componentDidMount () {
+    this._updateErrors()
+  }
+
+  _handleChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    const newState = {...this.state, table: evt.target.value}
+    this.setState(newState)
+    this._updateErrors(newState)
```

Comment:
Could you see if you could invoke `this._updateErrors` from `componentDidUpdate` (that gets called after a state is updated), so we don't need to pass the entire state to `this.setState`?

Reply:
Yep, that's a bit cleaner. updated in https://github.com/ibi-group/datatools-ui/pull/950/commits/d605d34f593aad5a7840da03a4470d46c108b960